malariagen_data/anoph/hap_freq.py

```diff
@@ -0,0 +1,391 @@
+from typing import Optional  # , Dict, Union, Callable, List
+
+import pandas as pd
+import numpy as np
+import xarray as xr
+import allel
+import dask.array as da
+from numpydoc_decorator import doc  # type: ignore
+
+from ..util import check_types, haplotype_frequencies
+from .hap_data import AnophelesHapData
+from .sample_metadata import locate_cohorts
+from . import base_params, frq_params  # , map_params, plotly_params
+
+
+class AnophelesHapFrequencyAnalysis(
+    AnophelesHapData,
+):
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        # N.B., this class is designed to work cooperatively, and
+        # so it's important that any remaining parameters are passed
+        # to the superclass constructor.
+        super().__init__(**kwargs)
+
+        # Set up cache variables.
+        self._cache_annotator = None
+
+    @check_types
+    @doc(
+        summary="""
+            Compute haplotype frequencies for a region.
+        """,
+        returns="""
+            A dataframe of haplotype frequencies, one row per haplotype.
+        """,
+        notes="""
+            Cohorts with fewer samples than `min_cohort_size` will be excluded from
+            output data frame.
+        """,
+    )
+    def haplotypes_frequencies(
+        self,
+        region: base_params.region,
+        cohorts: base_params.cohorts,
+        sample_query: Optional[base_params.sample_query] = None,
```

Comment:
Add `sample_query_options` parameter and pass through internally where appropriate.